### ISSUE TITLE
TripletMarginMiner does not allow normalization

### ISSUE THREAD

**Issue by marijnl** [KevinMusgrave/pytorch-metric-learning#88]
I think in this case it makes sense to support normalize_embeddings for TripletMarginMiner 
```
miner = miners.TripletMarginMiner(margin=0.1)
loss_func = losses.TripletMarginLoss(margin=0.1, normalize_embeddings=True)
hard_pairs = miner(embeddings, labels)
loss = loss_func(embeddings, labels, hard_pairs)
```

**Comment by KevinMusgrave** [comment]
All losses and miners have a ```normalize_embeddings``` parameter, which is set to ```True``` by default. So the same is true for TripletMarginMiner, i.e. embedding are normalized by default. You can see where it gets normalized [in BaseMiner](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/src/pytorch_metric_learning/miners/base_miner.py#L33)

**Comment by KevinMusgrave** [comment]
If you're using the logging provided by HookContainer, you can verify this by looking at 2 of the log files: ```tuple_miner_TripletMarginMiner.csv``` and ```metric_loss_TripletMarginLoss.csv```. In the metric loss log, you'll see the ```avg_embedding_norm``` is always 1, meaning the embeddings are normalized for the loss function. You can also see the number of triplets that actually contribute to the loss. Because you're using the same margin and normalizing embeddings in both the miner and loss, this number will always match the ```num_triplets``` column in the tuple miner log file. 

If you were normalizing embeddings in the loss but not the miner, then these two numbers would not match.

**Comment by KevinMusgrave** [comment]
As a side note, I can easily add ```avg_embedding_norm``` as one of the recordable attributes to the miners, if that helps people in debugging.

**Comment by marijnl** [comment]
Nice! 
I also see my mistake now. Sorry for the inconvenience.
Instead of calling the miner like this

`
hard_pairs = miner(embeddings, labels)
`

I do this:

`
hard_pairs = miner.mine(embeddings, labels)
`

This was causing some confusion on my end. 
I haven't really looked into the logging from the `HookContainer`, but that should be sufficient to see if normalization causes issues somewhere.
Thanks again
